distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering.enso

```diff
@@ -146,52 +142,37 @@ type Comparable
     has_custom_comparator : Atom -> Boolean
     has_custom_comparator atom =
         comp = Comparable.from atom
-        (comp.is_a Default_Unordered_Comparator).not && (comp.is_a Default_Ordered_Comparator).not
+        (comp.is_a Default_Comparator).not
 
 ## Singleton denoting that values of certain type are not comparable.
 type Incomparable
     Singleton
 
-## Default implementation of unordered comparator.
+## Default implementation of a _comparator_.
 @Builtin_Type
-type Default_Unordered_Comparator
-    is_ordered = False
-
-    equals : Any -> Any -> (Boolean|Nothing)
-    equals x y = Comparable.equals_builtin x y
-
-    hash : Any -> Integer
-    hash object = Comparable.hash_builtin object
-
-
-## Default implementation of an ordered _comparator_. Handles only primitive types,
-   does not handle atoms, or vectors. Any type that requires an ordering, must
-   define its own ordered comparator.
-@Builtin_Type
-type Default_Ordered_Comparator
-    is_ordered = True
-
-    ## Handles only primitive types, not atoms or vectors.
+type Default_Comparator
     compare : Any -> Any -> (Ordering|Nothing)
     compare x y =
-        case Comparable.less_than_builtin x y of
+        case Comparable.equals_builtin x y of
+            True -> Ordering.Equal
+            # If equals_builtin returns Nothing, x and y are incomparable and thus, it makes
+            # no sense to try it passing into less_than_builtin
```

Comment:
why change the order (`True` before `Nothing`) then ?

Reply:
No particular reason, I have changed the order once again, so that `Nothing` is the very first cas ebranch.